lib/Backend/Sym.cpp

```diff
@@ -987,117 +996,133 @@ Sym::Dump(IRDumpFlags flags, const ValueType valueType)
-                if (this->AsStackSym()->IsVar())
-                {
-                    if (this->AsStackSym()->HasObjectTypeSym() && !SimpleForm)
-                    {
-                        Output::Print(_u("<s%d>"), this->AsStackSym()->GetObjectTypeSym()->m_id);
-                    }
-                }
-                else
-                {
-                    StackSym *varSym = this->AsStackSym()->GetVarEquivSym(nullptr);
-                    if (varSym)
-                    {
-                        Output::Print(_u("(s%d)"), varSym->m_id);
-                    }
-                }
-                if (!SimpleForm)
-                {
-                    if (this->AsStackSym()->m_builtInIndex != Js::BuiltinFunction::None)
-                    {
-                        Output::Print(_u("[ffunc]"));
-                    }
-                }
-            }
-        }
-        if(IsStackSym())
-        {
-            IR::Opnd::DumpValueType(valueType);
-        }
-    }
-
-    if (this->IsPropertySym())
-    {
-        PropertySym *propertySym = this->AsPropertySym();
-
-        if (!SimpleForm)
-        {
-            Output::Print(_u("("));
-        }
-
-        Js::ScriptContext* scriptContext;
-        switch (propertySym->m_fieldKind)
-        {
-        case PropertyKindData:
-        {
-            propertySym->m_stackSym->Dump(flags, valueType);
-            scriptContext = propertySym->m_func->GetScriptContext();
-            Js::PropertyRecord const* fieldName = scriptContext->GetPropertyNameLocked(propertySym->m_propertyId);
-            Output::Print(_u("->%s"), fieldName->GetBuffer());
-            break;
-        }
-        case PropertyKindSlots:
-        case PropertyKindSlotArray:
-            propertySym->m_stackSym->Dump(flags, valueType);
-            Output::Print(_u("[%d]"), propertySym->m_propertyId);
-            break;
-        case PropertyKindLocalSlots:
-            propertySym->m_stackSym->Dump(flags, valueType);
-            Output::Print(_u("l[%d]"), propertySym->m_propertyId);
-            break;
-        default:
-            AssertMsg(0, "Unknown field kind");
-            break;
-        }
-
-        if (!SimpleForm)
-        {
-            Output::Print(_u(")"));
-        }
-    }
+	else if (this->IsStackSym())
+	{
+		StackSym *stackSym = this->AsStackSym();
+
+		if (stackSym->IsArgSlotSym())
+		{
+			if (stackSym->m_isInlinedArgSlot)
+			{
+				Output::Print(_u("iarg%d"), stackSym->GetArgSlotNum());
+			}
+			else
+			{
+				Output::Print(_u("arg%d"), stackSym->GetArgSlotNum());
+			}
+			Output::Print(_u("(s%d)"), m_id);
+		}
+		else if (stackSym->IsParamSlotSym())
+		{
+			if (stackSym->IsImplicitParamSym())
+			{
+				switch (stackSym->GetParamSlotNum())
+				{
+				case 1:
+					Output::Print(_u("callInfo"));
+					break;
+				case 2:
+					Output::Print(_u("funcInfo"));
+					break;
+				case 3:
+					Output::Print(_u("genObj"));
+					break;
+				case 4:
+					Output::Print(_u("genFrame"));
```

Comment:
Same thing as the generator frame.

---
In reply to: [75061008](https://github.com/Microsoft/ChakraCore/pull/1373#discussion_r75061008) [](ancestors = 75061008)